src/react/components/Highlight.tsx

```diff
@@ -18,18 +18,30 @@
 
 import React from 'react';
 
-const Highlight = ({ children, word }) => {
+interface Props {
+  children: string;
+  word?: string;
+}
+
+const Highlight = ({ children, word }: Props): React.ReactElement => {
   if (!word) {
-    return children;
+    return <>{children}</>;
   }
 
-  return children.split(word).reduce((prev, current, index) => [
-    prev,
-    <span key={index} className="bg-warning text-dark">
-      {word}
-    </span>,
-    current,
-  ]);
+  return (
+    <>
+      {children.split(word).map((chunk, index) => (
```

Comment:
what about this?
```
children.split(new RegExp(`(${word})`, 'g')).map((chunk, index) => (
  <React.Fragment key={index}>
    { chunk === word ? ( 
      <span className="bg-warning text-dark">{chunk}</span>
     ) : (
       <span>{chunk}</span>
     )}
  </React.Fragment>
))
```
splitting on a regex does not remove the word from the array, so you don't need to nest fragments

Reply:
Be aware of symbols in RegExp, they lead to unexpected behaviors. For instance both `x` and `.` are not contained in the text `123`, but:

```javascript
const word = "x";
"123".split(new RegExp(`(${word})`, 'g'))  // returns ["123"]
```

```javascript
const word = ".";
"123".split(new RegExp(`(${word})`, 'g'))  // returns ["", "1", "", "2", "", "3", ""]
```

Reply:
Correct, let's just leave it as it it. Right now i can't think of anything simpler